plugins/tor/tor_

```diff
@@ -0,0 +1,491 @@
+                sys.exit(-1)
+            else:
+                routers = response.split('\n')
+	        onr = 0
+		for router in routers:
+		    if router[0] == "r":
+	               onr += 1 
+                
+                print('routers.value {}'.format(onr))
+
+
+class TorTraffic(TorPlugin):
+    def __init__(self):
+        pass
+
+    def conf(self):
+        graph = {'title': 'Traffic',
+                 'args': '-l 0 --base 1024',
+                 'vlabel': 'data',
+                 'category': 'Tor',
+                 'info': 'bytes read/written'}
+        labels = {'read': {'label': 'read', 'min': 0, 'type': 'DERIVE'},
+                  'written': {'label': 'written', 'min': 0, 'type': 'DERIVE'}}
+
+        TorPlugin.conf_from_dict(graph, labels)
+
+    def fetch(self):
+        with gen_controller() as controller:
+            try:
+                authenticate(controller)
+            except stem.connection.AuthenticationFailure as e:
+                print('Authentication failed ({})'.format(e))
+                return
+
+            response = controller.get_info('traffic/read', None)
+            if response is None:
+                print("Error while reading traffic/read from Tor daemon", file=sys.stderr)
+                sys.exit(-1)
+
+            print('read.value {}'.format(response))
+
+            response = controller.get_info('traffic/written', None)
+            if response is None:
+                print("Error while reading traffic/write from Tor daemon", file=sys.stderr)
+                sys.exit(-1)
+            print('written.value {}'.format(response))
+
+
+##########################
+# Main                   
+##########################
+
+
+def main():
+    if len(sys.argv) > 1:
+        param = sys.argv[1].lower()
+    else:
+        param = 'fetch'
+
+    if param == 'autoconf':
+        TorPlugin.autoconf()
+        sys.exit()
+    elif param == 'suggest':
+        TorPlugin.suggest()
+        sys.exit()
+    else:
+        # detect data provider
+        if __file__.endswith('_bandwidth'):
+            provider = TorBandwidth()
+        elif __file__.endswith('_connections'):
+            provider = TorConnections()
+        elif __file__.endswith('_countries'):
+            provider = TorCountries()
+        elif __file__.endswith('_dormant'):
+            provider = TorDormant()
+        elif __file__.endswith('_flags'):
+            provider = TorFlags()
+        elif __file__.endswith('_routers'):
+            provider = TorRouters()
+        elif __file__.endswith('_traffic'):
+            provider = TorTraffic()
+        else:
+            print('Unknown plugin name, try "suggest" for a list of possible ones.')
+            sys.exit()
+
+        if param == 'config':
+            provider.conf()
+        elif param == 'fetch':
+            provider.fetch()
+        else:
+            print('Unknown parameter "{}"'.format(param))
```

Comment:
This message should be redirected to stderr and followed by a non-zero exitcode.